sdk/translation/azure-ai-translation-document/tests/test_all_document_statuses.py

```diff
@@ -90,40 +56,163 @@ def test_list_statuses_with_pagination(self, client):
-        translation_inputs = [
-            DocumentTranslationInput(
-                source_url=source_container_sas_url,
-                targets=[
-                    TranslationTarget(
-                        target_url=target_container_sas_url,
-                        language_code=target_language
-                    )
-                ]
-            )
-        ]
-
         # submit and validate job
-        job_id = self._submit_and_validate_translation_job(client, translation_inputs, len(blob_data))
+        job_id = self._create_translation_job_with_dummy_docs(client, docs_count, language_code=target_language, wait=False)
 
         # check doc statuses
         doc_statuses = list(client.list_all_document_statuses(job_id=job_id, skip=skip))
-        self.assertEqual(len(doc_statuses), docs_len - skip)
+        self.assertEqual(len(doc_statuses), docs_count - skip)
 
         # iterate over docs
         for document in doc_statuses:
-            self._validate_doc_status(document, target_language)
\ No newline at end of file
+            self._validate_doc_status(document, target_language)
+
+
+    @DocumentTranslationPreparer()
+    @DocumentTranslationClientPreparer()
+    def test_list_submitted_jobs_filter_by_status(self, client):
+        docs_count = 10
+        target_language = "es"
+
+        # submit and validate job
+        job_id = self._create_translation_job_with_dummy_docs(client, docs_count, language_code=target_language, wait=False)
+
+        # list jobs
+        statuses = ["Running"]
+        doc_statuses = list(client.list_all_document_statuses(job_id, statuses=statuses))
+        self.assertIsNotNone(doc_statuses)
+
+        # check statuses
+        for document in doc_statuses:
+            self._validate_doc_status(document, target_language, status=statuses)
+
+
+    @DocumentTranslationPreparer()
+    @DocumentTranslationClientPreparer()
+    def test_list_submitted_jobs_filter_by_ids(self, client):
+        docs_count = 5
+        target_language = "es"
+
+        # submit and validate job
+        job_id = self._create_translation_job_with_dummy_docs(client, docs_count, language_code=target_language, wait=False)
+
+        # filter ids
+        doc_statuses = list(client.list_all_document_statuses(job_id)) # convert from generic iterator to list
+        self.assertEqual(len(doc_statuses), docs_count)
+        ids = [doc.id for doc in doc_statuses]
+        ids = ids[:docs_count//2]
+
+        # do the testing
+        doc_statuses = list(client.list_all_document_statuses(job_id, ids=ids))
+        self.assertEqual(len(doc_statuses), len(ids))
+        for document in doc_statuses:
+            self._validate_doc_status(document, target_language, ids=ids)
+
+
+    @DocumentTranslationPreparer()
+    @DocumentTranslationClientPreparer()
+    def test_list_submitted_jobs_filter_by_created_start(self, client):
+        # it's not practical to test for created time filter
```

Comment:
you can delete this then :)

Reply:
sure, i was just leaving them in case you had something in mind on how to test these